frontend/ui/lib/model/trip.dart

```diff
@@ -1,9 +1,18 @@
 import 'package:tripmeout/model/location.dart';
 
 class Trip {
-  Trip({this.name, this.id, this.location});
+  Trip({this.name = "", this.id = "", location})
+      : location = location ?? Location();
 
   final String name;
   final String id;
   final Location location;
+
+  static Trip from(Trip trip, {name, id, location}) {
+    return Trip(
+      name: name ?? trip.name,
```

Comment:
I'm guessing that `""` will always be selected over `trip.name` here and this only does the right thing if `name` is null?

Reply:
Yeah, but non-positional arguments are optional here. The goal of this function is to provide clone/override functionality all in one. So essentially:
```dart
Trip trip1 = Trip(name: "my trip", id: "1");
// trip1 == { name: "my trip", id: "1", location: { latitude: 0, longitude: 0 } }
Trip otherTrip1 = Trip.from(trip1);
// otherTrip1 == { name: "my trip", id: "1", location: { latitude: 0, longitude: 0 }  }
Trip trip2 = Trip.from(trip1, id: "2");
// trip2 == { name: "my trip", id: "2", location: { latitude: 0, longitude: 0 }  }
```

This brings up an important point though, that I should write some tests for this...

Reply:
Got it. I also just realized that `name` here is resolved from the static method param rather than an existing trip entry (in which case `null` is presumably the default).

I was actually going to mention that I don't usually like to add tests to "dumb" value classes but I'd feel compelled to do so in this case mostly because I don't know dart. Then I figured you probably knew what you were doing and I'd defer to you. 😉